include/pms-utils/misc/readfile.hpp

```diff
@@ -0,0 +1,22 @@
+#include <cerrno>
+#include <filesystem>
+#include <fstream>
+#include <optional>
+#include <string>
+
+namespace [[gnu::visibility("default")]] pms_utils {
+namespace misc::_internal {
+
+std::optional<std::string> try_readfile(const std::filesystem::path &path) {
```

Comment:
header function needs to be `inline`. Why not define this internally?
As this is not a generic, just more the entire thing to lib/ and accordingly don't install it

Reply:
i just figured it was small enough that it wasn't worth the hassle